samcli/commands/check/resources/LambdaFunction.py

```diff
@@ -0,0 +1,84 @@
+"""
+Class object for Lambda Functions. Contains object from template, as well as all data for lambda functions,
+excluding pricing info
+"""
+from typing import List
+from samcli.lib.providers.provider import Function
+from samcli.commands.check.resources.TemplateResource import TemplateResource
+
+
+class LambdaFunction(TemplateResource):
+    def __init__(self, resource_object: Function, resource_type: str):
+        super().__init__(resource_object, resource_type)
+        self._duration = -1
+        self._tps = -1
+        self._parents: List = []
+        self._children: List = []
+        self._number_of_requests = -1
+        self._average_duration = -1
+        self._allocated_memory = -1
+        self._allocated_memory_unit = ""
+
+    @property
+    def number_of_requests(self) -> int:
+        return self._number_of_requests
+
+    @number_of_requests.setter
+    def number_of_requests(self, num: int):
+        self._number_of_requests = num
+
+    @property
+    def average_duration(self) -> int:
+        return self._average_duration
+
+    @average_duration.setter
+    def average_duration(self, avg: int):
+        self._average_duration = avg
+
+    @property
+    def allocated_memory(self) -> int:
+        return self._allocated_memory
+
+    @allocated_memory.setter
+    def allocated_memory(self, mry: int):
+        self._allocated_memory = mry
+
+    @property
+    def allocated_memory_unit(self) -> str:
+        return self._allocated_memory_unit
+
+    @allocated_memory_unit.setter
+    def allocated_memory_unit(self, unit: str):
+        self._allocated_memory_unit = unit
+
+    @property
+    def children(self) -> List:
+        return self._children
+
+    @children.setter
+    def children(self, child_node: "LambdaFunction"):
+        self._children.append(child_node)
```

Comment:
Consider this example:
```
lamdba_function.children = child_a
lambda_function.children = child_b
print(lambda_function.children) # [child_a, child_b]
```

It's a bit not natural to me to use a setter like this. Because `.children` is already returning a list, I would expect I would just use `lambda_function.children.append(child_a)` to add a new child. And the setter is used overwrite `self._children` with another list of children.

Reply:
I changed `def parent` as well to this style